### ISSUE TITLE
Prestashop 1.7.x PHPSESSID cookie lifetime should be changed

### ISSUE THREAD

**Issue by 01Kuzma** [PrestaShop/PrestaShop#29312]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Describe the solution you'd like

Prestashop _PHPSESSID_ session cookie has a default expiration time set to 2075y.

That's not a good solution for a GDPR policy.

It could be checked in the online [demo](https://demo.prestashop.com/#/en/front)

This cookie lifetime should be set for a session only or there should be an option to change it.

### Do you plan to work on this feature?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by HanaRebaiQA** [comment]
Hello,

Thank you for your suggestion. The Maintainer Team will take it into consideration for future developments. 

Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [contacting a professional](https://www.prestashop-project.org/support/) to help you create it. 

If you or anyone else is willing to [contribute this feature](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/) to the PrestaShop Project, the teams will be happy to help by ensuring that it fits well within the project. Please let us know if that is the case!

Thank you

**Comment by Prestaworks** [comment]
The issue is in 
src/Core/Session/SessionHandler.php
session_set_cookie_params 
where lifetime is set to $this->lifetime it should be 0 for sessions.
it seems that lifetime is the unix timestamp, but PHP expects seconds so the session cookie gets a very long lifetime.

The quick fix is to just change
'lifetime' => $this->lifetime,
to 
'lifetime' => 0,

**Comment by boo-code** [comment]
This is a unit mismatch rather than a missing feature, and the option asked for already exists.

`config/config.inc.php:159` turns the configured lifetime into an absolute Unix timestamp, which is what `Cookie` wants, and line 180 then hands that same value to `SessionHandler`, whose `init()` passes it to `session_set_cookie_params(['lifetime' => ...])` (`src/Core/Session/SessionHandler.php:78`), which takes a duration in seconds. A timestamp read as a duration is what produces the far-future date.

Measured on 9.2.x, `Set-Cookie` for `PHPSESSID` on a front page:

| PS_COOKIE_LIFETIME_FO | PHPSESSID |
| --------------------- | --------- |
| 480 (default) | `expires=Fri, 04-Jun-2083; Max-Age=1790534531` |
| 0 | no `expires`, no `Max-Age` - a session cookie |

The `PrestaShop-*` cookies in the same response expire on time, `Max-Age=1728000`, which is the 20 days the 480 hours are meant to mean.

So "set for a session only" is already reachable today by setting the Front Office cookie lifetime to 0, and the "option to change it" is that same setting. What was broken is every non-zero value, which was applied as a moment in time instead of a length of time.

The same defect was reported again as #42432. #42716 fixes it by giving the duration its own variable and deriving the absolute moment from it, so both call sites keep their own meaning, and it now closes this issue as well.

**Comment by Prestaworks** [comment]
There's a difference between the php session cookie and prestashop frontend cookie. The session cookie should be removed when the browser is closed.

**Comment by boo-code** [comment]
Agreed, and that distinction is the reason the current code cannot express what you are describing.

`config/config.inc.php:157-160` reads the configured lifetime in hours and, when it is above zero, converts it to an absolute timestamp:

```php
$cookie_lifetime = (int) Configuration::get('PS_COOKIE_LIFETIME_FO');
if ($cookie_lifetime > 0) {
    $cookie_lifetime = time() + (max($cookie_lifetime, 1) * 3600);
}
```

That value is correct for `new Cookie(...)` at lines 164-175, which wants an expiry timestamp. The same variable is then handed to `SessionHandler` at line 180, and `init()` puts it into `session_set_cookie_params(['lifetime' => ...])` (`src/Core/Session/SessionHandler.php:78`), which takes a duration in seconds. So one variable feeds two APIs that read it in different units, which is what produces the 2083 expiry on `PHPSESSID`.

It also means the two cookies cannot be configured apart today. `PS_COOKIE_LIFETIME_FO = 0` does make `PHPSESSID` a session cookie, but it makes `ps-s<id>` one as well, since the same zero reaches the `Cookie` constructor - so it is not a way to get the behaviour you describe while keeping the front office cookie persistent.

Passing the duration to `SessionHandler` and the timestamp to `Cookie` is what separates them, and only after that does "the session cookie is removed when the browser closes" become expressible at all.

**Comment by Prestaworks** [comment]
I still think we should use PHP default behaviour on the session.
$session_cookie_lifetime = 0;
